### PR TITLE
Add database indices

### PR DESCRIPTION
**Pull request by Sybrand**
Added indices to all the columns we filter and join on.

### CODE REVIEW

**Review by Kyubinhan — APPROVED** [review]

**Review by conbrad — APPROVED** [review]
LGTM. Any idea what type of indices are generated by these commands (hash, btree)? I glanced at the docs and didn't see anything mentioned. I assume the `unique=true` parameter hints at a hash index. 

If we are able to force btree indices for columns that are typically part of range queries maybe we would notice a speed. `psql` should tell us what indices were created.

**Comment by Sybrand** [comment]
> LGTM. Any idea what type of indices are generated by these commands (hash, btree)? I glanced at the docs and didn't see anything mentioned. I assume the `unique=true` parameter hints at a hash index.
> 
> If we are able to force btree indices for columns that are typically part of range queries maybe we would notice a speed up. `psql` should tell us what indices were created.

Good question.
https://www.postgresql.org/docs/12/indexes-types.html
Looks like b-tree. "By default, the CREATE INDEX command creates B-tree indexes, which fit the most common situations."

I don't know about the indexes on the geometries.

**Comment by conbrad** [comment]
If we start experiencing more slowness it may be worth exploring postgis indices for geometries which use r-trees. 

https://postgis.net/workshops/postgis-intro/indexing.html
https://postgis.net/docs/performance_tips.html

**Comment by Sybrand** [comment]
> If we start experiencing more slowness it may be worth exploring postgis indices for geometries which use r-trees.
> 
> https://postgis.net/workshops/postgis-intro/indexing.html
> https://postgis.net/docs/performance_tips.html
Cool.

Re. the first link - so we're already there. If you look at the code alembic spat out for the geom index:
```
op.create_index('idx_prediction_model_grid_subsets_geom', 'prediction_model_grid_subsets',
                    ['geom'], unique=False, postgresql_using='gist')
```

File: api/alembic/versions/004ac48ffe18_indices.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+down_revision = '81c96876355a'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic ###
+    op.create_index(op.f('ix_hourly_actuals_rh_valid'), 'hourly_actuals', ['rh_valid'], unique=False)
+    op.create_index(op.f('ix_hourly_actuals_station_code'), 'hourly_actuals', ['station_code'], unique=False)
+    op.create_index(op.f('ix_hourly_actuals_temp_valid'), 'hourly_actuals', ['temp_valid'], unique=False)
```
Comment (andrea-williams) [inline api/alembic/versions/004ac48ffe18_indices.py:23]: why is it only `temp_valid` and `rh_valid` that are getting indices? Why not the `_valid`s for wind speed & dir, precip, etc.?

File: api/Makefile [hunk 2]
```diff
@@ -201,3 +201,6 @@ docker-run-marvin:
 	# Run a python script to download hourly actuals from BC Wild Fire on docker.
 	# Needs to run "make docker-run" first
 	docker-compose exec api python -m app.fireweather_bot.hourly_actuals
+
+database-upgrade:
+	PYTHONPATH=. $(POETRY_RUN) alembic upgrade head
```
[the remark this answers is not in the document]
Comment (Sybrand) [reply api/Makefile:206]: @andrea-williams do you want one? Up to now I've just been running these manually.

[the remark this answers is not in the document]
Comment (Sybrand) [reply api/Makefile]: @andrea-williams ok - added.

**Review by andrea-williams — APPROVED** [review]
Oops, sorry Sy - forgot to set this to Approved!

**Pull request merged**